components/serialization/json/src/main/java/com/microsoft/kiota/serialization/JsonParseNode.java

```diff
@@ -31,13 +29,36 @@
 /** ParseNode implementation for JSON */
 public class JsonParseNode implements ParseNode {
     private final JsonElement currentNode;
+    protected final Gson gson;
 
     /**
      * Creates a new instance of the JsonParseNode class.
      * @param node the node to wrap.
+     * @deprecated use {@link #JsonParseNode(JsonElement, Gson)} instead.
      */
+    @Deprecated
```

Comment:
```suggestion
    */
```

Perfectly fine to want the default.